src/megaapi_impl.cpp

```diff
@@ -12819,30 +12819,41 @@ bool MegaApiImpl::sync_syncable(Sync *sync, const char *name, string *localpath)
 
 void MegaApiImpl::sync_auto_resume_result(const string& localPath, const handle remoteNode, const long long localFp, const std::vector<std::string>& regExp, const error e)
 {
-    if (e != 0)
-    {
-        return;
-    }
-
     const int nextTag = client->nextreqtag();
-
     MegaSyncPrivate *sync = new MegaSyncPrivate(localPath.c_str(), remoteNode, -nextTag);
     sync->setLocalFingerprint(localFp);
 
-    if (!regExp.empty())
+    MegaRequestPrivate *request = new MegaRequestPrivate(MegaRequest::TYPE_ADD_SYNC);
+    request->setNodeHandle(remoteNode);
+    request->setNumber(localFp);
+    request->setTag(nextTag);
+    requestMap[nextTag] = request;
+
+    if (e == 0)
     {
-        auto re = make_unique<MegaRegExp>();
-        for (const auto& v : regExp)
+        if (!regExp.empty())
         {
-            re->addRegExp(v.c_str());
+            auto re = make_unique<MegaRegExp>();
+            for (const auto& v : regExp)
+            {
+                re->addRegExp(v.c_str());
+            }
+            sync->setRegExp(re.get());
         }
-        sync->setRegExp(re.get());
+
+        Sync *s = client->syncs.back();
+        s->appData = sync;
+        sync->setState(s->state);
+        syncMap[-nextTag] = sync;
     }
+    else
+    {
+        sync->setState(SYNC_FAILED);
+    }
+
+    fireOnRequestFinish(request, MegaError(e));
```

Comment:
Don't need to create a `MegaRequest` and notify here. Simply report the event on a callback, so the app is aware of the _sync resume result_ for this `sync` :)

Reply:
Which is what we're doing in the next line. However, for this version I'd like to keep the `fireOnRequestFinish` so that I don't have to change much in megasync. Otherwise it seems there's a lot of immediate changes necessary and I need to get on with one-way-sync.

Reply:
We can put this PR/task on hold until you have time to focus back on it. I dislike the idea of adding temporary changes for a feature that presumably will take a while to be completed (so those temporary changes may end up in the code base forever). It's very common that changes in the SDK require changes in the app and the SDK cannot be merged until the apps have been adapted and are ready to merge as well. This is just one more case :)